### PR TITLE
Lottery

### PR DESCRIPTION
**Pull request by AlexFox86**
Add lottery.md

### CODE REVIEW

**Review by AndrePanin — APPROVED** [review]

File: docs/developing-contracts/examples/lottery.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    Winners(BTreeMap<u32, ActorId>),
+    Players(BTreeMap<u32, Player>),
+    Balance(u128),
+    LotteryState(LotteryState),
+}
+```
+
+## Functions
+
+Lottery contract interacts with ERC20 contract through function `transfer_tokens`.
```
Comment (LouiseMedova) [inline docs/developing-contracts/examples/lottery.md:100]:
```suggestion
Lottery contract interacts with fungible token contract through function `transfer_tokens`.
```

File: docs/developing-contracts/examples/lottery.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+	to: &ActorId, /// - the recipient address
+	amount_tokens: u128 /// - the amount of tokens
+)
+```
+
+This function sends a message (the action is defined in the enum `FTAction`) and gets a reply (the reply is defined in the enum `FTEvent`).
+
+```rust
+let _transfer_response: FTEvent = msg::send_and_wait_for_reply(
+    self.token_address.unwrap(), /// - the ERC-20 contracts address
```
Comment (LouiseMedova) [inline docs/developing-contracts/examples/lottery.md:115]:
```suggestion
    self.token_address.unwrap(), /// - the fungible token contract address
```

File: docs/developing-contracts/examples/lottery.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+Lottery winner calculation. Only the owner can pick the winner.
+
+```rust
+async fn pick_winner(
+	&mut self
+)
+```
+
+These functions are called in `async fn main()` through enum `LtAction`.
```
Comment (LouiseMedova) [inline docs/developing-contracts/examples/lottery.md:152]: I think it also worth mentioning that it is the program entry point and the program expects a message in the format `LTAction`.

**Review by LouiseMedova — APPROVED** [review]

File: docs/developing-contracts/examples/lottery.md (hunk 3 above)
Comment (AlexFox86) [reply docs/developing-contracts/examples/lottery.md:152]: Fixed on:
These functions are called in `async fn main()` through enum `LtAction`.
This is the entry point to the program, and the program is waiting for a message in `LtAction` format.

**Pull request merged**